worker/remoterelations/remoterelations.go

```diff
@@ -369,23 +418,106 @@ func (w *relationUnitsWatcher) loop() error {
 				continue
 			}
 			logger.Debugf("relation units changed: %#v", change)
-			if err := w.updateRelationUnitsChange(change); err != nil {
+			if evt, err := w.relationUnitsChangeEvent(change); err != nil {
 				return errors.Trace(err)
+			} else {
+				if evt == nil {
+					continue
+				}
+				event = *evt
+				changes = w.changes
 			}
+		case changes <- event:
+			changes = nil
 		}
 	}
 }
 
-func (w *relationUnitsWatcher) updateRelationUnitsChange(
+func (w *relationUnitsWatcher) relationUnitsChangeEvent(
 	change watcher.RelationUnitsChange,
-) error {
-	// TODO(wallyworld)
-	return ObserverRelationUnitsChange(change)
+) (*params.RemoteRelationChangeEvent, error) {
+	logger.Debugf("update relation units for %v", w.relationTag)
+	if len(change.Changed)+len(change.Departed) == 0 {
+		return nil, nil
+	}
+	// Ensure all the changed units have been exported.
+	changedUnitNames := make([]string, 0, len(change.Changed))
+	for name := range change.Changed {
+		changedUnitNames = append(changedUnitNames, name)
+	}
+	unitNamesToExport := append(changedUnitNames, change.Departed...)
+	logger.Debugf("exporting units: %v", unitNamesToExport)
```

Comment:
I suggest moving this to where we actually call the export API, otherwise this could get quite noisy